ontoconv/pipelines.py

```diff
@@ -183,6 +196,162 @@ def generate_pipeline(
     }
 
 
+def generate_ontoflow_pipeline(
+    ts: Triplestore,
+    resources: dict,
+    recognised_keys: "Optional[Union[dict, str]]" = "basic",
+) -> dict:
+    """Return a declarative ExecFlow pipeline as a dict.
+
+    Arguments:
+        ts: Tripper triplestore documenting data sources and sinks.
+        resources: A dict referring to OTEAPI partial pipelines for
+            a set of data sources and sinks.
+            See the example below for the expected structure.
+        client_iri: IRI of OTELib client to use.
+
+    Returns:
+        Dict-representation of a declarative ExecFlow pipeline.
+
+    Examples:
+        Example of the `resources` argument:
+
+        ```python
+        {'sinks': [
+           {'iri': 'ss3:AbaqusConfiguration',
+            'resourcetype': 'ss3:AbaqusSimulation'},
+           {'iri': 'ss3:AluminiumMaterialCard',
+            'resourcetype': 'ss3:AbaqusSimulation'},
+           {'iri': 'ss3:ConcreteMaterialCard',
+            'resourcetype': 'ss3:AbaqusSimulation'}],
+         'sources': [
+           {'iri': 'ss3kb:abaqus_config1', 'resourcetype': 'dataset'},
+           {'iri': 'ss3:AluminiumMaterialCard',
+            'resourcetype': 'ss3:MaterialCardGenerator'},
+           {'iri': 'ss3kb:abaqus_materialcard_concrete1',
+            'resourcetype': 'dataset'}]}
+        ```
+    """
+    names = {"input": [], "output": []}
+    strategies = []
+
+    lst = [("output", s) for s in resources.get("sources", [])]
+    lst.extend([("input", s) for s in resources.get("sinks", [])])
+    i = 1
+    for dtype, dct in lst:
+        iri = dct["iri"]
+        resourcetype = dct["resourcetype"]
+        suffix = (
+            iri.split("#", 1)[-1] if "#" in iri else iri.rsplit("/", 1)[-1]
+        )
+        if resourcetype == "dataset":
+            resource = load_container(ts, iri, recognised_keys=recognised_keys)
+        else:
+            r = load_simulation_resource(ts, resourcetype)
+            try:
+                resource = r[dtype][iri]
+            except KeyError:
+                try:
+                    resource = r[dtype][ts.prefix_iri(iri)]
+                except KeyError as exc:
+                    raise KeyError(
+                        f"Could not find {dtype} {iri} in {r['iri']}"
```

Comment:
Instead of the final `{r['iri']}` we would ideally like to add the name of the simulation tool. But that is not available here. My suggestion is just to remove `{r['iri']}`.

Reply:
We do actuall have the name of the resourcetool, but I had chosen the wrong variable. Updated.